pyiron_base/jobs/flex/pythonfunctioncontainer.py

```diff
@@ -65,7 +66,9 @@ def set_input(self, *args, **kwargs):
             **kwargs: Keyword arguments to be passed to the function.
         """
         self.input.update(
-            inspect.signature(self._function).bind(*args, **kwargs).arguments
+            DataContainer(
+                inspect.signature(self._function).bind(*args, **kwargs).arguments
+            ).to_builtin()
```

Comment:
@pmrv This is a quick workaround to check if the object can be stored in a data container. The previous code - just calling `update()` - did not fail when we set a dictionary with integer keys, while this workaround fails with a nice error message. Still it might be nice to have a higher level function to do this check instead.

Reply:
The problem is not quite clear to me, so I don't know if it could be checked before hand.
In principle DataContainer can handle integer keys, but it expects them to be in ascending order, if that was the issue here.

Reply:
I guess the difference is between `update(wrap=False)` and `update(wrap=True)` - with the wrap option is raises a `ValueError` when I provide a dict like `{1: 33, 2: 151, 3: 6}`.